File: C#/Algorithms/Recursion/Factorial.cs
/**
 * Class Factorial
 */
public class Factorial {

    /**
     * Calculates the factorial of a number using recursion
     *
     * @param int number
     * @return int
     */
    public int factorialRecursion(int number) {
        if (number == 0) {
            return 1;
        }

        return number * factorialRecursion(number - 1);
    }
}

File: Java/Algorithms/Math/Factorial.java
/**
 * Class Factorial
 */
public class Factorial {

    /**
     * Calculates the factorial of a number using iteration
     *
     * @param number a given number
     * @return int
     */
    public int factorialIteration(int number) {
        int factorial = 1;

        for (int i = 2;i < number + 1;i++) {
            factorial *= i;
        }

        return factorial;
    }
}
